core/src/main/scala/io/finch/Endpoints.scala

```diff
@@ -337,11 +338,13 @@ trait Endpoints {
    * at least one element) multi-value query-string param `name` from the request into a `Seq` or
    * raises a [[Error.NotPresent]] exception when the params are missing or empty.
    */
-  def paramsNonEmpty(name: String): Endpoint[Seq[String]] =
+  def paramsNonEmpty(name: String): Endpoint[NonEmptyList[String]] =
```

Comment:
How do you feel about introducing a new function `paramsNel` and keeping the old one (behind deprecation)? I want to keep the API unchanged and I think `paramsNel` sounds like a reasonable name.